From 3e0114e0d6e354b3f7be99976240618a9890452c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?121=E7=8F=AD32=E8=99=9F=E8=95=AD=E6=BE=A7=E9=82=A6?=
 <shou692199@gmail.com>
Date: Wed, 29 Sep 2021 15:10:26 +0200
Subject: [PATCH] metalava memory limit up to 3112M It can work on low ram
 machine. Don't need to use -j1, just brunch <device> or mka.

Change-Id: Ic14bb16bd61624b086fe3fe68f4c5894f9534ecf
---
 java/config/config.go   | 2 +-
 java/config/makevars.go | 2 +-
 java/droiddoc.go        | 1 +
 3 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/java/config/config.go b/java/config/config.go
index 1d0dd617ef..0785022604 100644
--- a/java/config/config.go
+++ b/java/config/config.go
@@ -64,7 +64,7 @@ const (
 func init() {
 	pctx.Import("github.com/google/blueprint/bootstrap")
 
-	pctx.StaticVariable("JavacHeapSize", "2048M")
+	pctx.StaticVariable("JavacHeapSize", "1024M")
 	pctx.StaticVariable("JavacHeapFlags", "-J-Xmx${JavacHeapSize}")
 	pctx.StaticVariable("DexFlags", "-JXX:OnError='cat hs_err_pid%p.log' -JXX:CICompilerCount=6 -JXX:+UseDynamicNumberOfGCThreads")
 
diff --git a/java/config/makevars.go b/java/config/makevars.go
index b355fad871..cf05fa3716 100644
--- a/java/config/makevars.go
+++ b/java/config/makevars.go
@@ -42,7 +42,7 @@ func makeVarsProvider(ctx android.MakeVarsContext) {
 	ctx.Strict("COMMON_JDK_FLAGS", "${CommonJdkFlags}")
 
 	ctx.Strict("DX", "${D8Cmd}")
-	ctx.Strict("DX_COMMAND", "${D8Cmd} -JXms16M -JXmx2048M")
+	ctx.Strict("DX_COMMAND", "${D8Cmd} -JXms16M -JXmx1024M")
 	ctx.Strict("R8_COMPAT_PROGUARD", "${R8Cmd}")
 
 	ctx.Strict("TURBINE", "${TurbineJar}")
diff --git a/java/droiddoc.go b/java/droiddoc.go
index b564fea014..cdaf8a6cc1 100644
--- a/java/droiddoc.go
+++ b/java/droiddoc.go
@@ -1474,6 +1474,7 @@ func metalavaCmd(ctx android.ModuleContext, rule *android.RuleBuilder, javaVersi
 
 	cmd.BuiltTool(ctx, "metalava").
 		Flag(config.JavacVmFlags).
+		Flag("-J-Xmx3112m").
 		FlagWithArg("-encoding ", "UTF-8").
 		FlagWithArg("-source ", javaVersion.String()).
 		FlagWithRspFileInputList("@", srcs).
